src/Text/Pandoc/Parsing.hs

```diff
@@ -205,18 +208,30 @@ type Parser t s = Parsec t s
 
 type ParserT = ParsecT
 
-newtype F a = F { unF :: Reader ParserState a } deriving (Monad, Applicative, Functor)
+-- | Reader monad wrapping the parser state. This is used to possibly delay
+-- evaluation until all relevant information has been parsed and made available
+-- in the parser state.
+newtype Delayable s a = Delayable { runDelayed :: Reader s a }
```

Comment:
Thank you, I changed the name multiple times and still wasn't satisfied. *Future* is so much better.